pkg/utils/json.go

```diff
@@ -1,550 +1,53 @@
+ * You may obtain a copy of the License at
  *
- * Permission is hereby granted, free of charge, to any person obtaining a copy
- * of this software and associated documentation files (the "Software"), to deal
- * in the Software without restriction, including without limitation the rights
- * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
- * copies of the Software, and to permit persons to whom the Software is
- * furnished to do so, subject to the following conditions:
+ *     http://www.apache.org/licenses/LICENSE-2.0
  *
- * The above copyright notice and this permission notice shall be included in all
- * copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
- * SOFTWARE.
- *
- * The source code of this file is written based on json-iterator,
- * all modifications are Copyright 2021 CloudWeGo Authors.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
  */
 
 package utils
 
 import (
-	"errors"
-	"fmt"
-	"strings"
-	"unicode/utf16"
-	"unicode/utf8"
-	"unsafe"
-)
-
-// const of json keyword char
-const (
-	EmptyJSON  = "{}"
-	Comma      = ','
-	Colon      = ':'
-	DQuotation = '"'
-	LeftBrace  = '{'
-	RightBrace = '}'
+	"github.com/bytedance/sonic"
 )
 
-const (
-	t1 = 0x00 // 0000 0000
-	tx = 0x80 // 1000 0000
-	t2 = 0xC0 // 1100 0000
-	t3 = 0xE0 // 1110 0000
-	t4 = 0xF0 // 1111 0000
-	t5 = 0xF8 // 1111 1000
-
-	maskx = 0x3F // 0011 1111
-
-	rune1Max = 1<<7 - 1
-	rune2Max = 1<<11 - 1
-	rune3Max = 1<<16 - 1
-
-	surrogateMin = 0xD800
-	surrogateMax = 0xDFFF
-
-	maxRune   = '\U0010FFFF' // Maximum valid Unicode code point.
-	runeError = '\uFFFD'     // the "error" Rune or "Unicode replacement character"
-
-	hex = "0123456789abcdef"
-)
+var sonicConifg = sonic.Config{
+	EscapeHTML:     true,
+	ValidateString: true,
+}.Froze()
 
 // Map2JSONStr transform map[string]string to json str, perf is better than use json lib directly
 func Map2JSONStr(mapInfo map[string]string) (str string, err error) {
 	defer func() {
 		if r := recover(); r != nil {
-			if e, ok := r.(error); ok {
-				err = fmt.Errorf("Map2JSONStr panic: %w", e)
-			} else {
-				err = fmt.Errorf("Map2JSONStr panic: %+v", r)
-			}
+			str, err = _Map2JSONStr(mapInfo)
```

Comment:
Seems not same as before

Reply:
this is for recover. What you mean?

Reply:
It's already recovered in line 32, and in `_Map2JSONStr`, recover is called again.

Reply:
So what? just in case of only sonic panic

Reply:
OK I see. The naming convention is a bit confusing.